text/0032-access_command.md

```diff
@@ -0,0 +1,224 @@
+```
+
+An example output for a published dataset:
+```
+$ qri access info me/dataset
+can-edit:  true
+visible:   true
+
+You can edit this dataset.
+
+This dataset is visible. Other users may see it when you're online, and if
+pushed to the registry it will show up in feeds for other users to browse and
+pull.
+```
+
+An example for a dataset you _don't_ own:
+```
+$ qri access info ca-state-parks/park-features
+can-edit:  false
+visible:   true
+
+You cannot edit this dataset, attempting to save to its history will create a 
+fork.
+
+This dataset is visible. Other users may see it when you're online, and if
+pushed to the registry it will show up in feeds for other users to browse and
+pull.
+```
+
+`$ qri access info me/dataset` for now will only but it will be _very_ helpful for viewing dataset permsissions that control both who can access a dataset and what they can do with it.
+
+### Visible on registry push
+Pushing unencrypted datasets† to the registry _requires_ a dataset be visible. To keep the old behaviour of a one-liner publish, pushing to the registry will automatically run `$ qri access visible` on the user's behalf. The push command will present this as user feedback:
+
+```
+$ qri push me/some_dataset
+setting dataset access to visible... done.
+pushing 28 versions of chriswhong/some_dataset...
+X of XX blocks pushed for version 12 August 2019 (Qmss4h552h3j2h33)...
+```
+
+Remotes, on the other hand _can_ accept hidden datasets. Remote configuration will get a `requireVisible` configuration parameter to get the same registry-like behaviour. Pushing hidden datasets is aimed at the collaborator use-case, where a datasets are explicitly pushed to trusted colleagues. when a user is pushed a hidden dataset, it will show up when they run `qri list` locally, but this dataset won't be broadcast back to the network.
+
+† _encrypted datasets don't exist yet._
+
+
+### Visible means visible _everywhere_
+In the past we've had permissions tied to a dataset/remote combination. As an example I could "publish" to a registry and that registry would have it.
+
+This RFC proposes permissions like visibility _only apply to datasets_. If you `push` a visible dataset to remote A, and peer B downloads that dataset, it's also `visible` for peer B.
+
+Later when we add access control, if you say "this dataset can _only_ be accessed by peer B", that will apply to anyone with the dataset.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### The `Visible` field
+For some time we've modelled `published` as a boolean field outside the dataset itself, stored as a logbook operation. We've struggled with this field not capturing the problem of _where_ a dataset's been published. Re-conceiving the new `visible` field as a "permission bit" clarifies the intent of this field. Published now answers "can the existence of this dataset be broadcasted" for the _entire history_ of a dataset. 
```

Comment:
We're still storing changes to visibility using logbook entries, right?

Reply:
yes. I'll add a note